modules/integration/tests-integration/tests-mqtt/src/test/java/org/wso2/mb/integration/tests/mqtt/functional/util/ResourceAdminServiceClient.java

```diff
@@ -0,0 +1,462 @@
+												description, fetchURL, null, null);
+	}
+
+	public void uploadArtifact(String description, DataHandler dh)
+			throws ResourceAdminServiceExceptionException, RemoteException {
+		String fileName;
+		fileName = dh.getName().substring(dh.getName().lastIndexOf('/') + 1);
+		resourceAdminServiceStub.addResource("/" + fileName, MEDIA_TYPE_GOVERNANCE_ARCHIVE,
+											 description, dh, null, null);
+	}
+
+	public String addCollection(String parentPath, String collectionName,
+								String mediaType, String description)
+			throws ResourceAdminServiceExceptionException, RemoteException {
+		return resourceAdminServiceStub.addCollection(parentPath, collectionName, mediaType, description);
+	}
+
+
+	public void addSymbolicLink(String parentPath, String name,
+								String targetPath)
+			throws ResourceAdminServiceExceptionException, RemoteException {
+
+		resourceAdminServiceStub.addSymbolicLink(parentPath, name, targetPath);
+	}
+
+	public void addTextResource(String parentPath, String fileName,
+								String mediaType, String description, String content)
+			throws RemoteException, ResourceAdminServiceExceptionException {
+
+		resourceAdminServiceStub.addTextResource(parentPath, fileName, mediaType,
+												 description, content);
+	}
+
+	public void addResourcePermission(String pathToAuthorize,
+									  String roleToAuthorize,
+									  String actionToAuthorize, String permissionType)
+			throws RemoteException, ResourceAdminServiceResourceServiceExceptionException {
+
+		resourceAdminServiceStub.addRolePermission(pathToAuthorize, roleToAuthorize,
+												   actionToAuthorize, permissionType);
+
+	}
+
+	public String getProperty(String resourcePath, String key)
+			throws RemoteException, ResourceAdminServiceExceptionException {
+
+		return resourceAdminServiceStub.getProperty(resourcePath, key);
+
+	}
+
+	public MetadataBean getMetadata(String resourcePath)
+			throws RemoteException, ResourceAdminServiceExceptionException {
+
+		return resourceAdminServiceStub.getMetadata(resourcePath);
+	}
+
+	public ContentBean getResourceContent(String resourcePath)
+			throws RemoteException, ResourceAdminServiceExceptionException {
+
+		return resourceAdminServiceStub.getContentBean(resourcePath);
+
+	}
+
+	public ResourceData[] getResourceData(String resourcePath)
+			throws RemoteException, ResourceAdminServiceExceptionException {
+		String[] resourceArray = {resourcePath};
+
+		return resourceAdminServiceStub.getResourceData(resourceArray);
+
+
+	}
+
+	public String getHumanReadableMediaTypes() throws Exception {
+		try {
+			return resourceAdminServiceStub.getHumanReadableMediaTypes();
+		} catch (Exception e) {
+			String msg = "get human readable media type error ";
+			throw new Exception(msg, e);
+		}
+	}
+
+	public String getMimeTypeFromHuman(String mediaType) throws Exception {
+
+		try {
+			return resourceAdminServiceStub.getMimeTypeFromHuman(mediaType);
+		} catch (Exception e) {
```

Comment:
Is there any reason why we are catching an generic exception rather than a specific exception?